lib/prima_ex_logger.ex

```diff
@@ -83,66 +83,57 @@ defmodule PrimaExLogger do
   end
 
   @spec process_metadata(Logger.metadata(), settings()) :: map()
   defp process_metadata(metadata, settings) do
-    metadata
-    |> Keyword.merge(settings.metadata)
-    |> alter_opentelemetry_metadata(settings)
-    |> Keyword.drop(@ignored_metadata_keys ++ settings.ignored_metadata_keys)
-    |> to_printable(settings.metadata_serializers)
-  end
-
-  @spec alter_opentelemetry_metadata(Logger.metadata(), settings()) :: Logger.metadata()
-  def alter_opentelemetry_metadata(metadata, settings)
+    metadata =
+      metadata
+      |> Keyword.merge(settings.metadata)
+      |> Keyword.drop(@ignored_metadata_keys ++ settings.ignored_metadata_keys)
 
-  def alter_opentelemetry_metadata(metadata, settings) do
     no_otel_metadata = Keyword.drop(metadata, @opentelemetry_sdk_metadata_keys)
+    metadata = if :raw == settings.opentelemetry_metadata, do: metadata, else: no_otel_metadata
 
-    case settings.opentelemetry_metadata do
-      # No opentelemetry-related metadata at all
-      :none ->
-        no_otel_metadata
+    to_printable(metadata, settings.metadata_serializers)
+  end
 
-      # Leave the standard log metadata set by the opentelemetry SDK untouched.
-      # Not recommended when sending logs to datadog, as this results in really ugly metadata.
-      :raw ->
+  @spec add_opentelemetry_metadata(map(), Logger.metadata(), settings()) :: map()
+  defp add_opentelemetry_metadata(event, metadata, settings) do
+    otel_metadata =
+      try do
         metadata
+        |> Keyword.take(@opentelemetry_sdk_metadata_keys)
+        |> opentelemetry_metadata(settings.opentelemetry_metadata)
+      rescue
+        # If the otel metadata is not in the format we expect
+        # (for example because the application may be overwriting it with arbitrary values)
+        # we don't want the logger to break, but just return no opentelemetry metadata.
+        _ -> []
+      end
 
-      format when format in [:datadog, :opentelemetry_clean, :detailed] ->
-        otel_metadata =
-          try do
-            metadata
-            |> Keyword.take(@opentelemetry_sdk_metadata_keys)
-            |> opentelemetry_metadata(format)
-          rescue
-            # If the otel metadata is not in the format we expect
-            # (for example because the application may be overwriting it with arbitrary values)
-            # we don't want the logger to break, but just return no opentelemetry metadata.
-            _ -> []
-          end
-
-        Keyword.merge(no_otel_metadata, otel_metadata)
-    end
+    otel_metadata |> to_printable(settings.metadata_serializers) |> Enum.into(event)
   end
 
-  @spec opentelemetry_metadata(Logger.metadata(), :datadog | :opentelemetry | :detailed) ::
+  @spec opentelemetry_metadata(
+          Logger.metadata(),
+          :datadog | :opentelemetry | :detailed | :raw | :none
+        ) ::
           Keyword.t()
-  def opentelemetry_metadata(raw_otel_metadata, format)
```

Comment:
I think this empty `def` should still be put after `@spec` for Best Practice :tm:

Reply:
Makes sense. Done